### PR TITLE
Analytics Plugin Support

### PR DESCRIPTION
**Pull request by joemcelroy**
Enables Analytics support for Search UI.

- [x] Unit tests for code
- [x] Manually tested locally

### CODE REVIEW

File: packages/search-ui-analytics-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# search-ui-analytics-plugin
```
Comment (JasonStoltz) [inline packages/search-ui-analytics-plugin/README.md:1]: We don't have to do this now, but perhaps in a future PR.

We should briefly explain what Behavioral Analytics is, and note that, for instance this is an experimental feature of Elastic that is not yet available.

Note that doing so will start sending events to the Elastic server, and maybe enumerate what those events are.

It would be also be super helpful to document somewhere in our docs something about "Analytics" and how you can use middleware to subscribe and handle all of these various events.

File: packages/search-ui-analytics-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@elastic/search-ui-analytics-plugin",
+  "version": "1.17.1",
```
Comment (JasonStoltz) [inline packages/search-ui-analytics-plugin/package.json:3]: I guess we don't really have the ability to release this as a pre-release or anything?

File: packages/search-ui-elasticsearch-connector/package.json [hunk 3]
```diff
@@ -41,6 +41,6 @@
   "dependencies": {
     "@elastic/elasticsearch": "^8.2.1",
     "@elastic/search-ui": "1.17.1",
-    "@searchkit/sdk": "^3.0.0-canary.58"
+    "@searchkit/sdk": "^3.0.0"
```
Comment (JasonStoltz) [inline packages/search-ui-elasticsearch-connector/package.json:44]: Is there a functional consequence of this? If so, should we create a separate PR for it?

File: packages/search-ui-elasticsearch-connector/src/handlers/search/__tests__/Response.test.ts [hunk 4]
```diff
@@ -43,8 +43,8 @@ describe("Search - Response", () => {
         label: "test",
         type: "value",
         entries: [
-          { label: "labeltest", count: "10" },
-          { label: "label2", count: "20" }
+          { label: "labeltest", count: 10 },
```
Comment (JasonStoltz) [inline packages/search-ui-elasticsearch-connector/src/handlers/search/__tests__/Response.test.ts:46]: Also wondering if this should be in a separate PR.

File: packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { AutocompleteSuggestion } from "../..";
```
Comment (JasonStoltz) [inline packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts:1]: I think we need to document this in `api-core-actions.md`. Something we can do in a follow up PR.

**Review by JasonStoltz — APPROVED** [review]
I'm going to approve this so that I'm not holding you up, but please review my comments before merging.

Note that we could make the suggested documentation changes in a follow up PR.

File: packages/search-ui-analytics-plugin/package.json (hunk 2 above)
Comment (joemcelroy) [reply packages/search-ui-analytics-plugin/package.json:3]: Yeh we dont have the ability yet for a pre-release without difficulty.

File: packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts (hunk 5 above)
Comment (joemcelroy) [reply packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts:1]: Thanks for the spot. Will follow up tomorrow with documentation for that new action.

File: packages/search-ui-analytics-plugin/README.md (hunk 1 above)
Comment (joemcelroy) [reply packages/search-ui-analytics-plugin/README.md:1]: Agreed. Definitely, documentation is a future task when analytics becomes publicly accessible. Il follow up with a PR to document it elsewhere for now.

File: packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts (hunk 5 above)
Comment (joemcelroy) [reply packages/search-ui/src/__tests__/actions/trackAutocompleteSuggestionClickThrough.test.ts:1]: done

**Review by JasonStoltz — APPROVED** [review]
